include/ht_shadervariable.h

```diff
@@ -73,21 +73,21 @@ namespace Hatchit {
         template<>
         inline void HT_API ShaderVariableTemplate<int>::SetData(int data)
         {
+            m_data = new BYTE[sizeof(int)];
```

Comment:
Also if SetData is called on an already-initialized ShaderVariableTemplate, shouldn't we free m_data before we allocate again?